bad_data/README.md

```diff
@@ -22,3 +22,7 @@ These are files used for reproducing various bugs that have been reported.
 
 * PARQUET-1481.parquet: tests a case where a schema Thrift value has been
   corrupted
+* arrow_issue_41321.parquet: test case of https://github.com/apache/arrow/issues/41321
```

Comment:
Maybe try to unify the file naming in this directory? We already have `PARQUET-1481.parquet` (a JIRA reference) so perhaps something like `ARROW-GH-41321.parquet`? (related: https://github.com/apache/parquet-testing/pull/57)

Reply:
Let me edit it 🤔

Reply:
done